File: mobile/android/app/src/main/java/com/kufarcode/mobile/data/models/Ad.kt
package com.kufarcode.mobile.data.models

data class Ad(
    val id: String,
    val title: String,
    val price: String?,
    val imageUrl: String?,
    val description: String? = null,
    val cityCode: String? = null,
    val seasonCode: String? = null,
    val location: GeoPoint? = null
)

data class GeoPoint(
    val type: String = "Point",
    val coordinates: List<Double> = emptyList()
)

File: mobile/src/types/ad.ts
export interface AdLocation {
  type: 'Point';
  coordinates: [number, number];
}

export interface Ad {
  id: string;
  title: string;
  price?: number;
  currency?: string;
  cityCode?: string;
  imageUrl?: string;
  seasonCode?: string;
  location?: AdLocation;
}
